### ISSUE TITLE
Possibility to do stratification with multi-output multi-class (multi-target) data

### ISSUE THREAD

**Issue by bundit786** [trent-b/iterative-stratification#26]
Hi,
I have a multi-output multi-class (multi-target) dataset and would like to do data stratification before applying a learning algorithm.  Using `iterative_train_test_split` from skmultilearn library (```
from skmultilearn.model_selection import iterative_train_test_split
x_train, y_train, x_test, y_test = iterative_train_test_split(x, y, test_size = 0.1)
```) would turned out an error, it seems to be working only multi-label data (binary class, multi-output). So is it possible for this iterative-stratification module could make to support for multi-target dataset.
Thank you.

**Comment by trent-b** [comment]
Hi, unfortunately this package is not currently designed to handle multiclass-multioutput classification. I wonder though if you could maybe [one-hot encode](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.OneHotEncoder.html#sklearn.preprocessing.OneHotEncoder) your targets, feed them into the stratifier, keep track of the indices of the instances, and then, once they are split into folds, convert them back into the original target values since you've retained the indices. Just a thought. I'm sorry I couldn't be of more help.

**Comment by bablf** [comment]
This does work but I needed to use the [MultiLabelBinarizer](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.MultiLabelBinarizer.html)

**Comment by dyhan316** [comment]
Thank you @trent-b  and @bablf !  Could you share a sample code that uses  `https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.MultiLabelBinarizer.html` ?
